cf-net/cf-net.c

```diff
@@ -0,0 +1,507 @@
+/*
+   Copyright (C) CFEngine AS
+
+   This file is part of CFEngine 3 - written and maintained by CFEngine AS.
+
+   This program is free software; you can redistribute it and/or modify it
+   under the terms of the GNU General Public License as published by the
+   Free Software Foundation; version 3.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with this program; if not, write to the Free Software
+  Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA  02111-1307, USA
+
+  To the extent this program is licensed as part of the Enterprise
+  versions of CFEngine, the applicable Commercial Open Source License
+  (COSL) may apply to this file if you as a licensee so wish it. See
+  included file COSL.txt.
+*/
+
+//#include <unix.h>
+#include <logging.h>            // Log, LogSetGlobalLevel
+
+#include <man.h>                // ManPageWrite
+#include <client_code.h>        // ServerConnection
+#include <crypto.h>             // CryptoInitialize
+#include <addr_lib.h>           // ParseHostPort
+#include <net.h>                // SocketConnect
+#include <time.h>               // time_t, time, difftime
+#include <stat_cache.h>         // cf_remote_stat
+// TODO: MOVE addr_lib out of libpromises....
+/*
+#include <generic_agent.h>
+#include <known_dirs.h>
+#include <eval_context.h>
+#include <lastseen.h>
+#include <files_names.h>
+#include <promises.h>
+#include <conversion.h>
+#include <vars.h>
+#include <communication.h>
+#include <string_lib.h>
+#include <rlist.h>
+#include <scope.h>
+#include <policy.h>
+#include <audit.h>
+#include <connection_info.h>
+#include <loading.h>
+#include <expand.h>
+#include <files_hashes.h>
+*/
+#define CF_NET_VERSION "0.0.0"
+
+static bool cfnet_silent = false;
+#define nprint(fmt, args...); {if(!cfnet_silent) printf(fmt, ##args);}
```

Comment:
`nprint()` should go in the final version, it's just being used everywhere at the moment with no differentiation. Most of the uses are `Log(LOG_LEVEL_VERBOSE)`, some are `ERR` and some are just `printf()`s, for example when printing the result of a command.